src/config/spawn.ts

```diff
@@ -55,23 +56,23 @@ export const spawnLocal: T.SpawnConductorFn = async (player: Player, { handleHoo
       handleHook(handle)
     }
 
-    // TODO: revisit for non-legacy
-    const newPort = await (!env.legacy
-      ? awaitInterfaceReady(handle, player.name)
-      : getTrueInterfacePortLegacy(handle, player.name)
-    )
-
-    if (newPort) {
-      player._adminInterfacePort = newPort
-    }
+    await awaitInterfaceReady(handle, player.name)
 
     const conductor = new Conductor({
+      player,
       name,
-      kill: async (...args) => handle.kill(...args),
+      kill: async (...args) => {
```

Comment:
This is a little improvement that means we actually fully wait for the killing of the child process before leaving the Promise... very useful, when the `Player` is having its `await player.shutdown()` method called